R/homoSapiens/humanAtlas.r

```diff
@@ -31,34 +31,17 @@ rm(GSE137710FilenameMelanoma)
 rm(GSE137710FilenameSpleen)
 rm(humanGenesFile)
 
-appendGenes <- function(humanGenesVector, GSEMatrix)
-{
-  rownamesGSEMatrix <- rownames(GSEMatrix) #Get rownames from GSEMatrix (new GSE file)
-  
-  rowCountHumanGenes <- nrow(humanGenesVector) #Calculate number of rows from list of full human genes
-  rowCountNewGSEFile <- nrow(GSEMatrix) #Calculate number of rows of GSE matrix
-  
-  missing_rows <- setdiff(humanGenesVector, rownamesGSEMatrix) #Use setdiff function to figure out rows which are different/missing from GSE matrix
-  missing_rows #Display missing rows
-  
-  zeroExpressionMatrix <- matrix(0, nrow = length(missing_rows), ncol = ncol(GSEMatrix)) #Create a placeholder matrix with zeroes and missing_rows length as row length
-  #zeroExpressionMatrix[1:3, 1:3] #Check first three entries of zeroExpressionMatrix
-  dim(zeroExpressionMatrix) #Check dimensions of matrix
-  
-  rownames(zeroExpressionMatrix) <- missing_rows #Assign row names
-  colnames(zeroExpressionMatrix) <- colnames(GSEMatrix) #Assign column names
-  #zeroExpressionMatrix[1:3, 1:3] #Check col and row names assigned correctly
-  
-  fullMatrix <- rbind(GSEMatrix, zeroExpressionMatrix) #Bind GSEMatrix and zeroExpressionMatrix together
-  dim(fullMatrix) #Check dimensions of fullMatrix
-  
-  #Reorder matrix
-  fullMatrix <- fullMatrix[humanGenesVector, ] #Reorder fullMatrix to preserve gene order
-  #fullMatrix[1:10, 1:3] #Check reordering
-  return(fullMatrix) #Return fullMatrix
-}
+source("~/Reference-Matrix-Generation/R/utils/check.r")
+
+ref_mats <- list(GSE129933Matrix, GSE147405Matrix)
+
+is_counts <- lapply(ref_mats, function(x)
+  {
+    checkRawCounts(x) == "raw counts"
+  }
+)
+ref_mats <- ref_mats[unlist(is_counts)]
 
-ref_mats <- list(GSE129933, GSE147405)
 new_mats <- lapply(ref_mats, function(x)
   {
   as.matrix(x) %>% appendGenes(humanGenesVector = humanGenesVector, GSEMatrix = .)
```

Comment:
This will error out, as the function is now defined with mouseGenesVector as the first argument, see other comment.

Reply:
See commit 72b0074